src/Optimization/hiopKKTLinSysSparse.cpp

```diff
@@ -572,70 +602,106 @@ namespace hiop
 #ifdef HIOP_USE_STRUMPACK              
           hiopLinSolverIndefSparseSTRUMPACK *p = new hiopLinSolverIndefSparseSTRUMPACK(n, nnz, nlp_);
           p->setFakeInertia(neq + nineq);
-          linSys_ = p;        
+          linSys_ = p;
+          actual_lin_solver = "strumpack";
 #endif  // HIOP_USE_STRUMPACK        
         }
-      } else {
-        //we are on the GPU. Our first choice is always cuSolver
-#if defined(HIOP_USE_CUSOLVER)        
-        hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
-        auto verbosity = hovScalars;
-        nlp_->log->printf(verbosity,
-                          "KKT_SPARSE_XDYcYd linsys: alloc CUSOLVER size %d (%d cons) (safe_mode=%d)\n",
-                          n,
-                          neq+nineq,
-                          safe_mode_);
-        if(safe_mode_) {
-          verbosity  = hovWarning;
+
+        if(linSys_) {
+          nlp_->log->printf(hovWarning,
+                            "KKT_SPARSE_XDYcYd linsys: alloc'ed sparse solver '%s' for size %d (%d cons) (cpu)\n",
+                            actual_lin_solver.c_str(),
+                            n,
+                            neq+nineq);
         }
+      } else if(compute_mode == "hybrid") {
+        assert(false == safe_mode_);
+        /////////////////////////////////////////////////////////////////////////////////////////////
+        // hybrid compute mode
+        //
+        // We still allow for CPU linear solver, but in a different order than for cpu compute mode
+        /////////////////////////////////////////////////////////////////////////////////////////////
+
+        //our first choice is cuSolver on hybrid compute mode
+        assert(nullptr == linSys_);
+        if(linear_solver == "cusolver" || linear_solver == "auto") {
+#if defined(HIOP_USE_CUSOLVER)        
+          hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
+          nlp_->log->printf(hovScalars,
+                            "KKT_SPARSE_XDYcYd linsys: alloc CUSOLVER size %d (%d cons) (hybrid)\n",
+                            n,
+                            neq+nineq);
+          p->setFakeInertia(neq + nineq);
+          linSys_ = p;
```

Comment:
Remove fake inertia:
```suggestion
          hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
          nlp_->log->printf(hovScalars,
                            "KKT_SPARSE_XDYcYd linsys: alloc CUSOLVER size %d (%d cons) (hybrid)\n",
                            n,
                            neq+nineq);
          linSys_ = p;
```